documents/validator.go

```diff
@@ -300,6 +300,29 @@ func anchoredValidator(repo anchors.AnchorRepository) Validator {
 	})
 }
 
+// latestVersionValidator checks if the document is the latest version
+func latestVersionValidator(repo anchors.AnchorRepository) Validator {
+	return ValidatorFunc(func(_, model Model) error {
+		nextID, err := anchors.ToAnchorID(model.NextVersion())
+		if err != nil {
+			return errors.New("failed to get anchorID: %v", err)
+		}
+
+		var empty [anchors.DocumentRootLength]byte
+		var rootByte [anchors.DocumentRootLength]byte
+
+		root, _, err := repo.GetAnchorData(nextID)
+		if err != nil {
+			return errors.New("failed to get document anchor %s from chain: %v", nextID.String(), err)
+		}
+		rootByte = root
+		if rootByte != empty {
```

Comment:
check length instead of creating an empty slice

Reply:
the length is always 32 bytes

Reply:
I will rename into `empty` in `zeros`

Reply:
Capacity is 32 bytes

Reply:
If you are getting len=32 thats a different issue since it returns garbage.

Reply:
Solidity returns a 32 byte-array filled with zeros for a non-existing anchor

Reply:
ok I see we have a function that you could use already :) 
`utils.IsEmptyByteSlice(rootByte[:])`